packages/concerto-core/lib/decoratormanager.js

```diff
@@ -157,11 +194,35 @@ class DecoratorManager {
      * with respect to to decorator command set model
      * @param {boolean} [options.validateCommands] - validate the decorator command set targets. Note that
      * the validate option must also be true
+     * @param {boolean} [options.migrate] - migrate the decoratorCommandSet $class to match the dcs model version
      * @returns {ModelManager} a new model manager with the decorations applied
      */
     static decorateModels(modelManager, decoratorCommandSet, options) {
+
+        if (options?.migrate) {
+            // get the version of the decoratorCommandSet from its $class property
+            const inputVersion = ModelUtil.parseNamespace(ModelUtil.getNamespace(decoratorCommandSet.$class)).version;
+
+            // if its < the currect DCS_Version, rewrite the $class version to match the supported DCS_VERSION
+            if (semver.lt(inputVersion, DCS_VERSION)) {
```

Comment:
This could do strange things if `inputVersion` is not compatible with `DCS_VERSION`, right? e.g. v1.0.0 to v30.0.0

Reply:
This function is designed to support changes across minor versions only (hence the original naming). When we get in to the territory or migrating the model across major versions, a rewrite (or a conceptual reconsideration) on how we manage DCS will be necessary.

Reply:
Will add an exception to safeguard against the strangeness in the above scenario.